content/blog/dbplyr-2-5-0/index.Rmd

```diff
@@ -0,0 +1,86 @@
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+
+We're most pleased to announce the release of [dbplyr](http://dbplyr.tidyverse.org/) 2.5.0. dbplyr is a database backend for dplyr that allows you to use a remote database as if it was a collection of local data frames: you write ordinary dplyr code and dbplyr translates it to SQL for you.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("dbplyr")
+```
+
+This post focuses on the biggest change in dbplyr 2.5.0: improved syntax for tables nested inside schema and catalogs. As usual, this release also contains a ton of minor improvements to SQL generation, and I'd highly recommend skimming the [release notes](https://github.com/tidyverse/dbplyr/releases/tag/v2.5.0) to learn the details. 
+
+```{r setup}
+library(dbplyr)
+```
+
+## Referring to tables in a schema
+
+Historically, dbplyr has provided a bewildering array of options to specify a tableinside a schema inside a catalog:
+
+```{r}
+#| eval: false
+con |> tbl(ident_q("catalog_name.schema_name.table_name"))
+con |> tbl(sql("SELECT * FROM catalog_name.schema_name.table_name"))
+con |> tbl(in_catalog("catalog_name", "schema_name", "table_name"))
+con |> tbl(ident_q("catalog_name.schema_name"), "table_name")
+con |> tbl(sql("catalog_name.schema_name"), "table_name")
+```
+
+You can also use `DBI::Id()`, whose syntax has also evolved over time:
+
+```{r}
+#| eval: false
+con |> tbl(DBI::Id(database = "catalog_name", schema = "schema_name", table = "table_name"))
+con |> tbl(DBI::Id(catalog = "catalog_name", schema = "schema_name", table = "table_name"))
+con |> tbl(DBI::Id("catalog_name", "schema_name", "table_name"))
+```
+
+Many of these options were poorly supported (i.e. we would accidentally break them from time-to-time) and suffered from the lack of a holistic vision. This release aims to bring order to the chaos by providing a succinct new syntax for literal table identifiers: `I()`. This allows you to succinctly identify a table nested inside a schema or catalog:
+
+```{r}
+#| eval: false
+con |> tbl(I("catalog_name.schema_name.table_name"))
+con |> tbl(I("schema_name.table_name"))
+```
+
+`I()` is a base function, and you may be familiar from it from modelling, e.g. `lm(y ~ x + I(y * z))`. It performs a similar role for both dbplyr and modelling function: it tells the function to treat the argument as is, rather than quoting it in the case of dbplyr, or interpreting as an interaction in the case of `lm()`.
```

Comment:
```suggestion
`I()` is a base function, and you may be familiar with it from modelling, e.g. `lm(y ~ x + I(y * z))`. It performs a similar role for both dbplyr and modelling function: it tells the function to treat the argument as is, rather than quoting it in the case of dbplyr, or interpreting as an interaction in the case of `lm()`.
```